File: medium/number-of-lines-to-write-string.py
class Solution:
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        nOfLines = 1
        lineLen = 0
        
        ab = "abcdefghijklmnopqrstuvwxyz"
        
        m = { ab[i]: widths[i] for i in range(26) }
        
        for c in S:
            if lineLen + m[c] <= 100:
                lineLen += m[c]
            else:
                lineLen = m[c]
                nOfLines += 1
        return [nOfLines, lineLen]

File: easy/number-of-lines-to-write-string.swift
class Solution {
    func numberOfLines(_ widths: [Int], _ S: String) -> [Int] {
        var numLine = 1
        var width = 0

        for c in S {
            width = width + widths[Int(c.unicodeScalars.first!.value-97)]
            if width > 100 {
                width = widths[Int(c.unicodeScalars.first!.value-97)]
                numLine += 1
            }
        }
        return [numLine, width]
    }
}
